### ISSUE TITLE
channels_last doesn't work

### ISSUE THREAD

**Issue by Khoa-NT** [fepegar/torchio#580]
**🐛Bug**

Hi, I'm learning TorchIO and I found `channels_last` doesn't do anything with an input tensor (C, W, H, D).

**To reproduce**

```python
import numpy as np
import torchio as tio

print(f'{tio.__version__=}')

# Input is numpy array
array = np.random.rand(1, 256, 256, 159).astype('float32')
print('\nInput numpy array')
print(array.shape)

print('\ntio.ScalarImage with channels_last=False')
array_ScalarImage = tio.ScalarImage(tensor=array, channels_last=False)
print(array_ScalarImage.shape)

print('\ntio.ScalarImage with channels_last=True')
array_ScalarImage = tio.ScalarImage(tensor=array, channels_last=True)
print(array_ScalarImage.shape)
```

**Expected behavior**

I expected it to return the tensor (D, C, W, H) according to the [document](https://torchio.readthedocs.io/data/image.html#torchio.Image).

**Actual behavior**

```python-traceback
tio.__version__='0.18.39'

Input numpy array
(1, 256, 256, 159)

tio.ScalarImage with channels_last=False
(1, 256, 256, 159)

tio.ScalarImage with channels_last=True
(1, 256, 256, 159)
```

**System info**

Output of `python <(curl -s https://raw.githubusercontent.com/fepegar/torchio/master/print_system.py)`:

```
Platform: Linux-5.4.0-73-generic-x86_64-with-glibc2.10
TorchIO:  0.18.39
PyTorch:  1.7.1
NumPy:    1.20.3
Python:   3.8.5 (default, Sep  4 2020, 07:30:14)
[GCC 7.3.0]
```

**Comment by romainVala** [comment]
Hello

this is not a bug.

channels_last argument, is only used when your read image from disk ( nifti file) .
In nifti 4D file, the 4D dimension is often the last dimension. If we want this last dimension to be interpreted as the channel dimension, the argument channels_last = True, will just permute the 4 dimension, to be first (channel dimension in pytorch)

If you use tensor as input, then no need of this extra argument, just reorder your tensor first ...

**Comment by Khoa-NT** [comment]
Thank you for clarifying. 
I think it should be included in the document. 
   
Can I ask one more question?
> If you use tensor as input, then no need of this extra argument, just reorder your tensor first ...

Currently, I'm working with volume data (D, H, W). 
If I only want to use the augmentation application (torchio.Compose()), then in my dataset class, I have to expand the tensor's dim to (C, D, H, W) and then reorder into (C, H, W, D).
After the augmentation, I have to reorder again into (C, D, H, W) for input to my model (I don't include the batch channel for explicit)?

**Comment by romainVala** [comment]
yes we should add it more explicitly in the doc...

Yes the extra batch dimension is added by pytorch dataloader, and not in the torchio transform

The specific ordering of the spatial dimension (D, H, W) does not really matter, for all the transform .. whatever the spatial ordering is you get the same at the end of the transform ... it is more a matter for the network inputs, to ensure the image are always presented similarly
for MRI data, we use the transform `ToCanonical` in order to have all data slice in the same spatial axis ...

**Comment by Khoa-NT** [comment]
> The specific ordering of the spatial dimension (D, H, W) does not really matter, for all the transform .. whatever the spatial ordering is you get the same at the end of the transform ... it is more a matter for the network inputs, to ensure the image are always presented similarly

Thank you. I only work with connectome data so I don't familiar with MRI data. 
I learned a lot from your tutorials.
I will try with the spatial order (D, H, W) and report if I have any problems.

**Comment by fepegar** [comment]
Thanks, @Khoa-NT, @romainVala.

This code might help:

```python
In [1]: import torchio as tio

In [2]: import torch

In [3]: im = tio.ScalarImage(tensor=torch.rand(3, 5, 5, 5))

In [4]: im.save('im.nii')

In [5]: import nibabel as nii

In [6]: import nibabel as nib

In [7]: nib.load('im.nii').shape
Out[7]: (5, 5, 5, 1, 3)

In [8]: bad_shape = nib.load('im.nii').get_fdata().squeeze()

In [10]: bad_shape.shape
Out[10]: (5, 5, 5, 3)

In [11]: nib.Nifti1Image(bad_shape, torch.eye(4)).to_filename('bad.nii')

In [12]: tio.ScalarImage('bad.nii').shape
Out[12]: (5, 5, 5, 3)

In [13]: tio.ScalarImage('bad.nii', channels_last=True).shape
Out[13]: (3, 5, 5, 5)
```
